components/server/src/api/auth-provider-service-api.ts

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Code, ConnectError, HandlerContext, ServiceImpl } from "@connectrpc/connect";
+import { AuthProviderService as AuthProviderServiceInterface } from "@gitpod/public-api/lib/gitpod/v1/authprovider_connect";
+import { inject, injectable } from "inversify";
+import { PublicAPIConverter } from "@gitpod/gitpod-protocol/lib/public-api-converter";
+import {
+    CreateAuthProviderRequest,
+    CreateAuthProviderResponse,
+    GetAuthProviderRequest,
+    GetAuthProviderResponse,
+    ListAuthProvidersRequest,
+    ListAuthProvidersResponse,
+    ListAuthProviderDescriptionsRequest,
+    ListAuthProviderDescriptionsResponse,
+    UpdateAuthProviderRequest,
+    UpdateAuthProviderResponse,
+    DeleteAuthProviderRequest,
+    DeleteAuthProviderResponse,
+} from "@gitpod/public-api/lib/gitpod/v1/authprovider_pb";
+import { AuthProviderService } from "../auth/auth-provider-service";
+import { AuthProviderEntry, AuthProviderInfo } from "@gitpod/gitpod-protocol";
+import { Unauthenticated } from "./unauthenticated";
+
+@injectable()
+export class AuthProviderServiceAPI implements ServiceImpl<typeof AuthProviderServiceInterface> {
+    constructor(
+        @inject(PublicAPIConverter) private readonly apiConverter: PublicAPIConverter,
+        @inject(AuthProviderService) private readonly authProviderService: AuthProviderService,
+    ) {}
+
+    async createAuthProvider(
+        request: CreateAuthProviderRequest,
+        context: HandlerContext,
+    ): Promise<CreateAuthProviderResponse> {
+        const ownerId = request.owner.case === "ownerId" ? request.owner.value : undefined;
+        const organizationId = request.owner.case === "organizationId" ? request.owner.value : undefined;
+
+        if (!organizationId && !ownerId) {
```

Comment:
we could use uuid library to check for validate values? i.e. like in https://github.com/gitpod-io/gitpod/blob/c33b555448b0b566600282615402fb94cd72dd92/components/server/src/api/organization-service-api.ts#L183-L191